### PR TITLE
bugfix: update selected options for controlled and uncontrolled fields, update tests

### PR DESCRIPTION
**Pull request by mariia-kovsharova**
Selected values into the MultiSelect field can be provided after the component rendering, in this case we need to update the selected values for both controlled and uncontrolled components.

* Fixed the logic.
* Added tests for controlled and uncontrolled versions.
* Separated controlled and uncontrolled version tests into groups.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -20,6 +20,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/) and this p
     -   Added logic for contolled state for the component if `selectedItems` and `onSelection` passed.
 -   `<StickyNoteModal/>`
     -   static test id `data-test-id="sticky-note-modal"` will be removed with next major version
+        `<MultiSuggestField />`
```
Comment (haschek) [inline CHANGELOG.md:23]: Looks like the list item point is missing here, is it? So the auto-format does not work correctly.

File: src/components/MultiSuggestField/MultiSuggestField.stories.tsx [hunk 2]
```diff
@@ -78,6 +78,40 @@ predefinedNotControlledValues.args = {
...
+
+            <button onClick={() => setLoaded((prev) => !prev)}>Toggle selected</button>
+        </>
+    );
+};
+
+/**
+ *
+ */
+export const deferredSelection = DeferredSelectionTemplate.bind({});
```
Comment (haschek) [inline src/components/MultiSuggestField/MultiSuggestField.stories.tsx:113]: The "clear all" button on the field does not work for this example. Is it just the story, or is is some bug within the elemen code?

**Review by haschek — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (mariia-kovsharova) [reply CHANGELOG.md]: Yes, fixed the point, we had the same above - so I just added the current fix there

File: src/components/MultiSuggestField/MultiSuggestField.stories.tsx (hunk 2 above)
Comment (mariia-kovsharova) [reply src/components/MultiSuggestField/MultiSuggestField.stories.tsx:113]: That was a story bug, I forgot to remove the selection function from the component.
Check again please.

Also, I added tests with clearance for both controlled and uncontrolled components.

File: src/components/MultiSelect/MultiSelect.tsx [hunk 3]
```diff
@@ -448,7 +447,7 @@ export function MultiSelect<T>({
             <IconButton
                 disabled={disabled}
                 name="operation-clear"
-                data-test-id="clear-all-items" // @deprecated should be created from the given testid plus `_clearance` suffix
+                data-test-id={dataTestId ?? dataTestid ? `${dataTestId ?? dataTestid}_clearance` : undefined}
```
Comment (haschek) [inline src/components/MultiSelect/MultiSelect.tsx:450]: We need to postpone this change to the moment when we plan to release a new major version, for the upcoming minor version we need to keep the old test id.

**Review by haschek — CHANGES_REQUESTED** [review]

**Review by haschek — APPROVED** [review]

**Pull request merged**